cronjob_scripts/trigger-package-build.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+
+from __future__ import annotations
+
+if __name__ == "__main__" and __package__ is None:
+    import sys
+    from pathlib import Path
+
+    # Add repo root to PYTHONPATH and make relative imports work
+    sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+    __package__ = "scripts"
+
+
+from collections import Counter
+import json
+import os
+import random
+import subprocess
+import time
+
+import requests
+
+from cronjob_scripts.get_latest_version import CrateVersionDict, get_latest_version
+from cronjob_scripts.checkout_worktree import checkout_worktree_for_arch
+from cronjob_scripts.stats import get_requested_crates
+
+TARGET_ARCH_TO_BUILD_OS = {
+    "x86_64-apple-darwin": "macos-latest",
+    "aarch64-apple-darwin": "macos-latest",
+    "x86_64-unknown-linux-gnu": "ubuntu-20.04",
+    "x86_64-unknown-linux-musl": "ubuntu-20.04",
+    "x86_64-pc-windows-msvc": "windows-latest",
+    "aarch64-pc-windows-msvc": "windows-latest",
+    "aarch64-unknown-linux-gnu": "ubuntu-20.04",
+    "aarch64-unknown-linux-musl": "ubuntu-20.04",
+    "armv7-unknown-linux-musleabihf": "ubuntu-20.04",
+    "armv7-unknown-linux-gnueabihf": "ubuntu-20.04",
+}
+
+
+def main():
+    target_arches = get_target_architectures()
+    for target_arch in target_arches:
+        trigger_for_arch(target_arch)
+
+
+def trigger_for_arch(target_arch: str):
+    print(f"Triggering builds for {target_arch}")
+
+    build_os = get_build_os(target_arch)
+
+    tracking_worktree_path = checkout_worktree_for_arch(target_arch)
+    excludes = get_excludes(tracking_worktree_path, days=7, max_failures=5)
+
+    popular_crates = get_popular_crates()
+    requested_crates = get_requested_crates(period="1 day", arch=target_arch)
+
+    possible_crates = set(popular_crates + requested_crates) - set(excludes)
+
+    # check random crates up to a limit.
+    check_limit = int(os.environ.get("CRATE_CHECK_LIMIT", 5))
+    # Note that the old next-unbuilt-package.sh script actually checked the top 4 * CRATE_CHECK_LIMIT
+    # crates from get-stats.sh + popular-crates.txt rather than shuffling the whole lot before checking.
+    # The original script just looped through the whole list, but kept track of where it got to,
+    # to avoid getting stuck on unbuildable crates.
+    #
+    # For now I'm trying to be roughly backwards compatible with the old script's numbers,
+    # but I think that shuffling first is an important bugfixt. Once we are using crates.io sparse
+    # index api without rate-limits, we can think about checking more crates.
+    crates_to_check = random.sample(list(possible_crates), 4 * check_limit)
+    if os.environ.get("RECHECK"):
+        crates_to_check = ["cargo-quickinstall"] + crates_to_check
```

Comment:
```suggestion
        crates_to_check.append("cargo-quickinstall")
```

Reply:
it would be .insert(0, "cargo-quickinstall") - I want to check quickinstall first, before the other random crates, because this is used as part of the release process.